Zpp/Simulation/Agents/JobDistributor/JobDistributor.cs

```diff
@@ -0,0 +1,148 @@
+                if (ScheduleOperation(productionOrderOperation: first, machineId: machine.machineId.Value))
+                {
+                    OperationManager.RemoveOperation(first);
+                }
+            }
+        }
+
+        private void WithDrawMaterials(ProductionOrderOperation productionOrderOperation)
+        {
+            OperationManager.WithdrawMaterialsFromStock(operation: productionOrderOperation, TimePeriod);
+        }
+
+        private bool ScheduleOperation(ProductionOrderOperation productionOrderOperation, int machineId)
+        {
+            var machine = ResourceManager.GetResourceRefById(new Id(machineId));
+            if (machine.IsWorking)
+            {
+                Debug.WriteLine("Machine is still Working.");
+                return false;
+            }
+
+            machine.IsWorking = true;
+            var msg = Resource.Resource.Work.Create(productionOrderOperation, machine.ResourceRef);
+
+            if (productionOrderOperation.GetValue().Start <= TimePeriod)
+            {
+                _SimulationContext.Tell(message: msg, sender: Self);
+                return true;
+            } // else operation starts in the future and has to wait.
+
+            var delay = productionOrderOperation.GetValue().Start - TimePeriod;
+            Schedule(delay, msg);
+            return true;
+        }
+
+        private void CreateMachines(ResourceDictionary machineGroup, long time)
+        {
+            foreach (var machines in machineGroup)
+            {
+                foreach (var machine in machines.Value)
+                {
+                    var machineNumber = ResourceManager.Count + 1;
+                    var agentName = $"{machine.GetValue().Name}({machineNumber})".ToActorName();
+                    var resourceRef = Context.ActorOf(Resource.Resource.Props(_SimulationContext, time)
+                        , agentName);
+                    var resource = new ResourceDetails(machine, resourceRef);
+                    ResourceManager.AddResource(resource);
+                }
+            }
+        }
+
+        private void ProvideMaterial(ProductionOrderOperation operation)
+        {
+            // TODO Check for Preconditions (Previous job is finished and Material is Provided.)
+            var rawOperation = operation.GetValue();
+            rawOperation.ProducingState = ProducingState.Finished;
+            var machineId = rawOperation.MachineId;
+            if (machineId == null)
+                throw new Exception("Machine not found.");
+
+            OperationManager.InsertMaterialsIntoStock(operation, TimePeriod);
+
+            var machine = ResourceManager.GetResourceRefById(new Id(machineId.Value));
+            machine.IsWorking = false;
+
+            PushWork();
+        }
+
+        protected override void Finish()
+        {
+            Console.WriteLine(Sender.Path + " has been Killed");
```

Comment:
private static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
Logger.Info($"{Sender.Path} has been Killed");
--> my suggestion if you still want to use the logging backend, else it's fine.

Reply:
Changed it to Debug.WriteLine because it will be ignored with "Release" build.
I'd like to keep it that way until everything is working.